api/helpers/gamestate/moves/resolve/execute.js

```diff
@@ -59,6 +59,19 @@ module.exports = {
       case 1:
         result = sails.helpers.gamestate.moves.resolve.ace(result);
         break;
+      case 3:
+      case 4:
+      case 5:
+      case 7:
+        // These one-offs require an additional user input after resolving
+        // Each one has its own phase designated by the rank
+        result = {
+          ...result,
+          phase: oneOff.rank,
+          oneOff, // oneOff stays on the stack until next move
+          turn: result.turn - 1, // turn doesn't increment until the next move
```

Comment:
It also feels a little weird to increment the turn above and decremenet it here. Again, this comes down to being DRY. I don't see a way to neatly separate turn incrementing in one-offs ace, 2, 6, and 9 resolving, but not 3-5 + 7 (the multi phase one-offs), while also handling each one individually, without checking the rank twice and repeating the common changes e.g. the turn going up for the single-phase one-offs and the turn not going up for the multi-phase one-offs.

So my approach here is to assign the common values for all the single-stage one-offs up above and then in the switch statement call an individual rank-based helper to actually 'do' the one-off part of the single stage one-offs e.g. `ace()` to scrap all points and `six()` to scrap all jacks, while the multi-phase one-offs are all handled in one cascading switch case here since all of them simply keep the turn + oneOff the same and set the phase to their respective rank. Then the multi phase one-offs can all get their own moveType specific `execute()` and `validate()` helpers to handle their second stage e.g. `sevenPoints()`

Reply:
Yea, it's deff a little funky to decrement it, we could possible just increment it after the switch

```
if([1,2,6,9].includes(oneOff.rank){
result.turn++
}
```